src/gui/update.rs

```diff
@@ -757,15 +757,16 @@ pub fn handle_message(ajour: &mut Ajour, message: Message) -> Result<Command<Mes
 
                 addon.state = AddonState::Fingerprint;
 
-                let mut version = None;
+                // Set version & file id of installed addon to that of newly unpacked package.
                 if let Some(package) = addon.relevant_release_package(global_release_channel) {
-                    version = Some(package.version);
-                }
-                if let Some(version) = version {
-                    addon.set_version(version);
+                    addon.set_version(package.version);
+
+                    if let Some(file_id) = package.file_id {
+                        addon.set_file_id(file_id);
+                    }
                 }
```

Comment:
Here is where we set version / file id from the newly installed / updated package. This is done prior to updating the cache entry, so cache will always stay up to date with latest external release id when installing / updating